File: training-data/kotlin/TriangleClassify.kt
enum class TriangleKind { EQUILATERAL, ISOSCELES, SCALENE, INVALID }

fun classify(a: Double, b: Double, c: Double): TriangleKind {
    val (x, y, z) = listOf(a, b, c).sorted()
    if (x <= 0 || x + y <= z) return TriangleKind.INVALID
    return when (setOf(a, b, c).size) {
        1 -> TriangleKind.EQUILATERAL
        2 -> TriangleKind.ISOSCELES
        else -> TriangleKind.SCALENE
    }
}

fun main() {
    println(listOf(classify(3.0, 3.0, 3.0), classify(3.0, 4.0, 4.0), classify(3.0, 4.0, 5.0), classify(1.0, 1.0, 3.0)))
}

File: training-data/c/triangle_classify.c
#include <stdio.h>

const char *classify(double a, double b, double c) {
    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || b + c <= a) return "invalid";
    if (a == b && b == c) return "equilateral";
    if (a == b || b == c || a == c) return "isosceles";
    return "scalene";
}

int main(void) {
    printf("%s %s %s %s\n", classify(3, 3, 3), classify(3, 4, 4), classify(3, 4, 5), classify(1, 1, 3));
    return 0;
}
